app/models/user.rb

```diff
@@ -26,4 +27,8 @@ def save!; end
   def attributes
     super.merge(roles: nil, permissions: nil)
   end
+
+  def to_hash
+    instance_variables.map { |var| [var.to_s.delete('@'), instance_variable_get(var)] }.to_h
```

Comment:
is this related to the PR?

Reply:
Its been used for debugging a lot both this PR and the previous one.  Its useful to get a hash of the user as apposed to implementing a to_s method.  So I'd like to leave this in for now and flag it for removal once we're doing doing anything user related.

Reply:
ok, yeah, that's fine. I just didn't see the correlation with the timeout stuff